### PR TITLE
Fixes #879: IsChallengeCreator decorator in Challenge Phase detail API Endpoint.

### LINKED ISSUE #879: Use IsChallengeCreator decorator in Challenge Phase detail API Endpoint
Currently, the check for if the requesting user is Creator of Challenge or not is missing from the `challenge_phase_detail` API View. We would like to add the `IsChallengeCreator` decorator (https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/permissions.py#L6) to this endpoint. 

Also, please add test cases after making the above changes.

### PR DESCRIPTION
**Pull request by isht3**
* Add IsChallengeCreator decorator to challenge_phase_detail API.

    * Write tests for the following changes.

I've also added missing tests to challenge_detail end-point.

Fixes #879

### CODE REVIEW

**Comment by isht3** [comment]
@varunagrawal @deshraj Please take a look and let me know if the tests are enough. :smile:

File: tests/unit/challenges/test_views.py [hunk 1]
```diff
@@ -198,6 +198,54 @@ def test_get_particular_challenge(self):
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+    def test_put_non_creator_particular_challenge(self):
```
Comment (RishabhJain2018) [inline tests/unit/challenges/test_views.py]: Please change `test_put_non_creator_particular_challenge` -> `test_update_challenge_when_user_is_not_its_ceator`

File: tests/unit/challenges/test_views.py [hunk 2]
```diff
@@ -198,6 +198,54 @@ def test_get_particular_challenge(self):
...
+
+        self.client.force_authenticate(user=self.user1)
+
+        expected = {"detail": "Sorry, you are not allowed to perform this operation!"}
+
+        response = self.client.put(self.url, {'title': 'Rose Challenge', 'description': 'Version 2.0'})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_put_creator_particular_challenge(self):
```
Comment (RishabhJain2018) [inline tests/unit/challenges/test_views.py]: Please change `test_put_creator_particular_challenge` -> `test_update_challenge_when_user_is_its_ceator`

File: tests/unit/challenges/test_views.py [hunk 3]
```diff
@@ -1243,6 +1291,47 @@ def test_get_particular_challenge_phase(self):
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+    def test_put_non_creator_particular_challenge_phase(self):
```
Comment (RishabhJain2018) [inline tests/unit/challenges/test_views.py]: Please change `test_put_non_creator_particular_challenge_phase` -> `test_update_challenge_phase_when_user_is_not_its_ceator`

File: tests/unit/challenges/test_views.py [hunk 4]
```diff
@@ -1243,6 +1291,47 @@ def test_get_particular_challenge_phase(self):
...
+
+        self.client.force_authenticate(user=self.user1)
+
+        expected = {"detail": "Sorry, you are not allowed to perform this operation!"}
+
+        response = self.client.put(self.url, {'name': 'Rose Phase', 'description': 'New description.'})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_put_creator_particular_challenge_phase(self):
```
Comment (RishabhJain2018) [inline tests/unit/challenges/test_views.py]: Please change `test_put_non_creator_particular_challenge_phase` -> `test_update_challenge_phase_when_user_is_its_ceator`

**Comment by isht3** [comment]
@RishabhJain2018 Done!

**Pull request merged**